### ISSUE TITLE
Create a java client for the admin services

### ISSUE THREAD

**Issue by mandy-chessell** [odpi/egeria#284]
This enhancement requests the creation of a java client (or clients) for the Admin Services.  This should be tackled after #283.

**Comment by mandy-chessell** [comment]
This client is needed for FVT

**Comment by planetf1** [comment]
If we regard our REST api as external, documented, supported then the java client is one mechanism for testing - and of course beneficial to the java coder, but we may additionally test the REST api directly

**Comment by mandy-chessell** [comment]
We have not yet agreed that the REST API is public. but if we do then ok to test REST API independently

**Comment by mandy-chessell** [comment]
I have created an initial set of admin clients based around the server type.  This means there is a different client for each type of server, making it clear which configuration values to set up for each server type.

There are now clients for:
* Metadata Server
* Metadata Access Point
* Repository Proxy
* Conformance Test Server
* Discovery Server
* Stewardship Server

**Comment by planetf1** [comment]
We will also need to add to the assemblies - though currently only the full assembly is probably being used/tested.

**Comment by mandy-chessell** [comment]
The clients are added to the egeria-omag.xml as well as the client package - do we need to do more at this stage?  I understand we have a general problem with the client package in that it is not built with dependencies (yes?)

Is there something else?

**Comment by planetf1** [comment]
I don't see any changes to the .xml files in these commits? Is there another change to merge?

What I see in the clients & samples for the full assembly are:
```
➜  samples git:(master) pwd
/Users/jonesn/src/egeria/open-metadata-distribution/open-metadata-assemblies/target/egeria-1.5-SNAPSHOT-distribution/egeria-omag-1.5-SNAPSHOT/samples
➜  samples git:(master) ls
asset-management-samples-1.5-SNAPSHOT-jar-with-dependencies.jar
asset-management-samples-1.5-SNAPSHOT.jar
subject-area-client-samples-1.5-SNAPSHOT-jar-with-dependencies.jar
subject-area-client-samples-1.5-SNAPSHOT.jar
➜  samples git:(master) cd ..
ls
➜  egeria-omag-1.5-SNAPSHOT git:(master) ls
LICENSE           clients           samples           user-interface
NOTICE            conformance-suite server            utilities
➜  egeria-omag-1.5-SNAPSHOT git:(master) cd clients
ls
➜  clients git:(master) ls
README.md                                    discovery-engine-client-1.5-SNAPSHOT.jar
asset-catalog-client-1.5-SNAPSHOT.jar        governance-engine-client-1.5-SNAPSHOT.jar
asset-consumer-client-1.5-SNAPSHOT.jar       governance-program-client-1.5-SNAPSHOT.jar
asset-owner-client-1.5-SNAPSHOT.jar          information-view-client-1.5-SNAPSHOT.jar
community-profile-client-1.5-SNAPSHOT.jar    it-infrastructure-client-1.5-SNAPSHOT.jar
data-engine-client-1.5-SNAPSHOT.jar          project-management-client-1.5-SNAPSHOT.jar
data-platform-client-1.5-SNAPSHOT.jar        security-officer-client-1.5-SNAPSHOT.jar
data-privacy-client-1.5-SNAPSHOT.jar         software-developer-client-1.5-SNAPSHOT.jar
data-science-client-1.5-SNAPSHOT.jar         stewardship-action-client-1.5-SNAPSHOT.jar
dev-ops-client-1.5-SNAPSHOT.jar              subject-area-client-1.5-SNAPSHOT.jar
digital-architecture-client-1.5-SNAPSHOT.jar
```
The full set of client maven artifacts is:
```
  egeria find . -type d -name '*client*' | sort
./admin-services-client
./asset-catalog-client
./asset-consumer-client
./asset-lineage-client
./asset-owner-client
./community-profile-client
./data-engine-client
./data-platform-client
./data-privacy-client
./data-science-client
./design-model-client
./dev-ops-client
./digital-architecture-client
./discovery-engine-client
./discovery-engine-services-client
./gaf-metadata-client
./glossary-view-client
./governance-engine-client
./governance-program-client
./governance-program-client-samples
./information-view-client
./it-infrastructure-client
./ocf-metadata-client
./odf-metadata-client
./open-lineage-services-client
./open-metadata-client-package
./open-metadata-conformance-suite-client
./project-management-client
./repository-services-client
./rest-client-connectors
./rest-client-connectors-api
./rest-client-factory
./security-officer-client
./software-developer-client
./spring-rest-client-connector
./stewardship-action-client
./stewardship-engine-services-client
./stewardship-services-client
./subject-area-client
./subject-area-client-samples
./ui-admin-services-client
```

As can be seen above
 - The 'samples' do have dependencies pulled in - which makes sense as they can be launched directly
 - The 'clients' do not - this could make sense if we package & setup environment appropriately

I'd be inclined to
 *) Move samples to a seperate git repo (easier to clone, then can modify)

For the clients we could 
 a) leave the samples without dependencies, but somehow package them into a workable directory structure, with dependent libraries placed correctly in the filesystem
OR
 b) build each client independently with a 'with-dependencies' version - a user can chose which to use, and this version can be used directly.

**Comment by mandy-chessell** [comment]
The xml files already had the client modules included already so there was nothing to change

**Comment by mandy-chessell** [comment]
Later review of the assemblies uncovered that the directory name for the admin services is incorrect.  Fixed now.

**Comment by planetf1** [comment]
I think we still have more work to do here in the packaging as per the comments. (suggest master/1.6)

Though it could be a new Issue.

 - The only utility is open-metadata-types . This doesn't include dependencies so cannot simply be 'run'
 - The samples are no longer in the archive. We do have a new project 'egeria-dev-projects' - perhaps they'll go there? Shall we track that in #1643
 - The clients IMO offer very limited value in this archive -- they do not include prereqs, and in most cases the likely developer integration is through maven artifacts

**Comment by mandy-chessell** [comment]
These seem to be different issues to implementing the admin client - should be separate issues..
